src/vmm/src/builder.rs

```diff
@@ -498,7 +526,7 @@ pub fn build_microvm_from_snapshot(
 
     // Restore devices states.
     let mmio_ctor_args = MMIODevManagerConstructorArgs {
-        mem: guest_memory,
+        mem: guest_memory.clone(),
```

Comment:
It could, but I think there's a lot of things that need to change all over the `persist.rs` files of various components. I wouldn't want to do this in this particular PR.

Reply:
From what I can see, only changes needed are:
```rust
  // in MMIODevManagerConstructorArgs
  -pub mem: GuestMemoryMmap,
  +pub mem: &GuestMemoryMmap,

  // in MMIODeviceManager::restore
  -let mem = &constructor_args.mem;
  +let mem = constructor_args.mem;
```

Reply:
oh, you are (almost) right! I thought it would be more. Will do.